fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -871,6 +884,24 @@ public CommandProcessingResult createDatatable(final JsonCommand command) {
         return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withResourceIdAsString(datatableName).build();
     }
 
+    private void createIndexesForTable(String datatableName, JsonArray columns) {
+        for (final JsonElement column : columns) {
+            createIndexForColumn(datatableName, column.getAsJsonObject());
+        }
+    }
+
+    private void createIndexForColumn(String datatableName, JsonObject column) {
+        String name = column.has("name") ? column.get("name").getAsString() : null;
+        final Boolean unique = column.has("unique") ? column.get("unique").getAsBoolean() : false;
+        final Boolean indexed = column.has("indexed") ? column.get("indexed").getAsBoolean() : false;
+        if (indexed) {
+            if (!unique) {
```

Comment:
Can you pls explain why we don't create the index if the column is also marked as unique? Thanks.

Reply:
When unique constraint is added on a column, an index is created by database with the same name as unique constraint name. so no need to create index separately.